src/geo/map/cartodb-layer.js

```diff
@@ -112,8 +117,17 @@ var CartoDBLayer = LayerModelBase.extend({
     return this.get('source');
   },
 
-  setSource: function (source, options) {
-    this.set('source', source, options);
+  setSource: function (newSource, options) {
+    this._updateSourceReferences(newSource);
+    this.set('source', newSource, options);
```

Comment:
options? for wat?

Reply:
For example, for setting the source silently with:

```
layer.setSource(newSource, { silent: true });
```

We're currently using this [here](https://github.com/CartoDB/cartodb/blob/master/lib/assets/core/javascripts/cartodb3/deep-insights-integration/layers-integration.js#L267). We might want to find a better name for that option when we polish the public API more.